lib/workers/repository/update/branch/index.ts

```diff
@@ -811,13 +814,16 @@ export async function processBranch(
         }
       } else if (config.automerge) {
         logger.debug('PR is configured for automerge');
-        const prAutomergeResult = await checkAutoMerge(pr, config);
-        if (prAutomergeResult?.automerged) {
-          return {
-            branchExists,
-            result: BranchResult.Automerged,
-            commitSha,
-          };
+        if (!commitSha || config.ignoreTests) {
```

Comment:
```suggestion
        // skip automerge if there is a new commit since status checks aren't done yet
        if (!config.artifactErrors?.length && (!commitSha || config.ignoreTests)) {
```

i think we don't like to automerge on artifact errors?

Reply:
the code doesn't reach here with artifact errors

Reply:
im rolling this back

Reply:
@viceice after checking that the code goes into this section in case of artifact errors, i removed the check
https://github.com/renovatebot/renovate/blob/9562cf400928fd5756febf0a42105da03c362ade/lib/workers/repository/update/branch/index.ts#L767-L812